plot_radial_build.py

```diff
@@ -0,0 +1,150 @@
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+import matplotlib.colors
+import yaml
+import argparse
+
+def build_composition_string(composition, max_characters):
+    """
+    Assembles string from composition dict for use in radial build plot
+
+    Arguments:
+        composition (dict): "material name (str)":volume_fraction (float)
+
+    Returns:
+        comp_string (string): formatted string with material definition
+    """
+
+    comp_string = ''
+    for material, fraction in composition.items():
+        
+        mat_string = f'{material}: {round(fraction*100,3)}%, '
+        line_len =len(comp_string)-(comp_string+mat_string).rfind('\n')
+
+        if line_len > max_characters:
+            comp_string += '\n' + mat_string
+        else:
+            comp_string += mat_string
+
+    return comp_string[0:-2]
+
+def plot_radial_build(build, title, colors = None, 
+                      max_characters = 35, max_thickness = None, size = (8,4), 
```

Comment:
What if you set `max_thickness` to default to a very large number (1e6?) so that it's effectively infinite by default.  Then you don't need to check if it's defined later on and can simplify the bounds checking.